scan/tls_revocation.go

```diff
@@ -0,0 +1,50 @@
+package scan
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+
+	"golang.org/x/crypto/ocsp"
+)
+
+// TLSRevocation contains scanners testing certificate revocation.
+var TLSRevocation = &Family{
+	Description: "Scans determining certificate revocation",
+	Scanners: map[string]*Scanner{
+		"OCSPStaple": {
+			"Checks OCSP Staple if given",
+			ocspStapleScan,
+		},
+	},
+}
+
+// ocspStapleScan checks the hosts OSCP staple if one is returned.
+func ocspStapleScan(addr, hostname string) (grade Grade, output Output, err error) {
+	conn, err := tls.DialWithDialer(Dialer, Network, addr, &tls.Config{
+		ServerName: hostname,
+		RootCAs:    RootCAs,
+	})
+	if err != nil {
+		return
+	}
+	conn.Close()
+
+	connState := conn.ConnectionState()
+
+	if len(connState.OCSPResponse) == 0 {
```

Comment:
change this to `conn.OCSPResponse()` for pre Go 1.5 support. https://github.com/golang/go/blob/master/src/crypto/tls/conn.go#L1068